src/utils/VideoChannelUtils.ts

```diff
@@ -46,36 +55,76 @@ export const addVideoChannel = async (roomId: string, roomName: string) => {
     await WidgetUtils.addJitsiWidget(roomId, CallType.Video, "Video channel", true, roomName);
 };
 
-export const getConnectedMembers = (room: Room, connectedLocalEcho: boolean): Set<RoomMember> => {
+// Gets the members connected to a given video room, along with a timestamp
+// indicating when this data should be considered stale
+const getConnectedMembers = (room: Room, connectedLocalEcho: boolean): [Set<RoomMember>, number] => {
     const members = new Set<RoomMember>();
+    const now = Date.now();
+    let allExpireAt = Infinity;
 
     for (const e of room.currentState.getStateEvents(VIDEO_CHANNEL_MEMBER)) {
         const member = room.getMember(e.getStateKey());
-        let devices = e.getContent<IVideoChannelMemberContent>()?.devices ?? [];
+        const content = e.getContent<IVideoChannelMemberContent>();
+        let devices = content?.devices ?? [];
+        const expiresAt = content?.expires_ts ?? -Infinity;
+        // Ignore events with a timeout that's way off in the future
+        const inTheFuture = expiresAt - STUCK_DEVICE_TIMEOUT_MS * 5 / 4 > now;
```

Comment:
these math operations need to get some brackets so it's clear what order you're trying to get the runtime to use, instead of remembering what the JS runtime does for order of operations.